ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java

```diff
@@ -147,21 +147,34 @@ public void registerPartitionStateChangeListener(StateModelListenerType listener
   }
 
   @Override
-  public boolean setReplicaSealedState(ReplicaId replicaId, boolean isSealed) {
+  public boolean setReplicaSealedState(ReplicaId replicaId, ReplicaSealStatus replicaSealStatus) {
     if (!(replicaId instanceof AmbryReplica)) {
       throw new IllegalArgumentException(
           "HelixParticipant only works with the AmbryReplica implementation of ReplicaId");
     }
     synchronized (helixAdministrationLock) {
       DataNodeConfig config = getDataNodeConfig();
       String partitionId = replicaId.getPartitionId().toPathString();
-      boolean success = true;
-      if (!isSealed && config.getSealedReplicas().remove(partitionId)) {
-        logger.trace("Removing the partition {} from sealedReplicas list", partitionId);
-        success = dataNodeConfigSource.set(config);
-      } else if (isSealed && config.getSealedReplicas().add(partitionId)) {
-        logger.trace("Adding the partition {} to sealedReplicas list", partitionId);
-        success = dataNodeConfigSource.set(config);
+      boolean success = false;
+      switch (replicaSealStatus) {
+        case NOT_SEALED:
+          if (removePartitionFromSealedReplicasConfig(partitionId, config) && removePartitionFromPartiallySealedReplicasConfig(partitionId, config)) {
+            logger.trace("Removing partition {} from sealed lists to mark it as NOT_SEALED", partitionId);
+            success = dataNodeConfigSource.set(config);
```

Comment:
The code logic/design is changed.
If we want to set the replica to NOT_SEALED state, in the old code, only when the partition is at sealed state (config.getSealedReplicas().remove(partitionId) returns true), we will call dataNodeConfigSource.set(config).

In the current code, even the partition currently is already at NOT_SEALED state, we still call dataNodeConfigSource.set(config)
Will the un-necessary/repeated set call any problem?

Reply:
Fixed it.